### DISCUSSION TITLE
Is there anyway for client components to pass down props to server components?

### DISCUSSION THREAD

**Discussion by madhawk1296** [vercel/next.js#51060] [App Router]
```
// example 
<ClientComponent setState={setState} >
  <ServerComponent state={state} />
</ClientComponent
```

**Comment by icyJoseph** [comment]
Nope, not with Next.js' current implementation of React Server Components, though this person made a side project framework, where that is possible:

- https://github.com/dai-shi/waku
- YouTube video where it is explained https://youtu.be/ciQdQkT0bZk?t=974

Since everything is a Server Component, unless marked with `use client`, or if its part of a Client Component tree, in your example, I can assume that both `ClientComponent` and `ServerComponent`, are being composed within another Client Component, like this:

```jsx
'use client'; // this part I assume is what you have

// later on

return (
<ClientComponent setState={setState} >
  <ServerComponent state={state} />
</ClientComponent
);
```
 
The moment you import `ServerComponent` into the module, it starts to behave as a client component. And that'll lead to a number of issues, from Client Components not being able to be `async` functions, to lack of Node API presence on the browser, to secrets or sensitive data being part of the JS bundle, etc...
